### PR TITLE
Framework overview

### PR DESCRIPTION
**Pull request by danielaRiesgo**
## Summary ##

Adding framework overview.

## Known issues ##

Don't like the name "event delegate" (in the Delegates section) but couldn't think of any other.

### CODE REVIEW

**Comment by woloxci** [comment]
:+1: There are no linter violations.

File: FRAMEWORK-OVERVIEW.md [hunk 1]
```diff
@@ -2,5 +2,82 @@
...
+
 To handle those screens, the framework uses the [MVVM pattern](http://www.sprynthesis.com/2014/12/06/reactivecocoa-mvvm-introduction/).
+
+This framework is intended to be customizable for the user to adapt it to his/her app. For this, the framework uses the [Delegation](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Protocols.html#//apple_ref/doc/uid/TP40014097-CH25-ID276) design pattern, also used by Apple's UIKit framework, for example.
+
+Finally, the Authentication framework also uses a functional reactive programming framework ([ReactiveCocoa](https://github.com/ReactiveCocoa/ReactiveCocoa)) which the user will need to use for conforming certain protocols (see [Session Service](#session-service) below).
+
+## Basic Structure
+
+Following the [MVVM pattern](http://www.sprynthesis.com/2014/12/06/reactivecocoa-mvvm-introduction/), each screen has a View, a View Model and a View Controller, where the View Model is in charge of handling all logic in order to provide all information the view needs to display and the View Controller is in charge of binding the View Model's information to the View's elements and viceversa.
```
Comment (FranDepascuali) [inline FRAMEWORK-OVERVIEW.md]: I would delete the `and viceversa`

File: FRAMEWORK-OVERVIEW.md [hunk 2]
```diff
@@ -2,5 +2,82 @@
...
+
+and
+
+* [SignupController](Authentication/Signup/View/SignupController.swift) (fixed controller)
+* [SignupView](Authentication/Signup/View/SignupView.swift) (default view)
+* [SignupViewModel](Authentication/Signup/ViewModel/SignupViewModel.swift) (default view model)
+
+
+
+The Signup screen has a "child screen" that is used for showing the terms and services of the app. This screen is supposed to only show the terms and services to the user and give the possibility to return to sign up. Since it's basic and it's not central, its components are not customizable, just the content it is shown (although the content is an HTML that can use javascript).
```
Comment (FranDepascuali) [inline FRAMEWORK-OVERVIEW.md]: `just the content it is shown` -> `just the content it shows`

File: FRAMEWORK-OVERVIEW.md [hunk 3]
```diff
@@ -2,5 +2,82 @@
...
+* [LoginControllerTransitionDelegate](Authentication/Login/View/LoginControllerTransitionDelegate.swift)
+* [SignupControllerTransitionDelegate](Authentication/Signup/View/SignupControllerTransitionDelegate.swift)
+
+
+
+
+
+## Localization
+
+This framework incentivates the user to localize its app. Every text seen in the framework's screens is localized and to change it you only need to create a [Strings file](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/LoadingResources/Strings/Strings.html) and localize the key of the text element you want to change, withuot affecting the rest.
```
Comment (FranDepascuali) [inline FRAMEWORK-OVERVIEW.md]: without

**Comment by FranDepascuali** [comment]
👍

**Pull request merged**